universum/modules/structure_handler.py

```diff
@@ -198,8 +206,16 @@ def execute_steps_recursively(self, parent: Optional[Step], cfg: Configuration,
 
                     # Here pass_errors=False, because any exception while executing build step
                     # can be step-related and may not affect other steps
-                    self.run_in_block(self.execute_one_step, step_name, False,
-                                      item, step_executor, obj_a.critical)
+                    step_process = None
+                    try:
+                        step_process = self.run_in_block(self.execute_one_step, step_name, False,
+                                                         item, step_executor, obj_a.critical)
+                        if not item.background:
```

Comment:
I also kinda don't like the
```
if not item.background:
    self._run_artifact_collection_step(item.name, step_process.collect_artifacts)
<one line>
if not item.background:
    self._run_artifact_collection_step(item.name, e.step_process.collect_artifacts)
```
part. These lines seem a little to similar. Are we sure this cannot be fixed somehow?